app/controllers/course/assessment/submission/answer/controller.rb

```diff
@@ -2,7 +2,10 @@
 class Course::Assessment::Submission::Answer::Controller < \
   Course::Assessment::Submission::Controller
 
-  load_resource :answer, class: Course::Assessment::Answer.name, through: :submission
+  load_and_authorize_resource :answer, class: Course::Assessment::Answer.name, through: :submission,
```

Comment:
Why does this have to be declared here, can it be declared in the specific controller which implements `create_programming_files` and `destroy_programming_file`?

Reply:
The subsequent controller `Course::Assessment::Submission::Answer::Programming::Controller` loads the `Course::Assessment::Answer::Programming` object through the answer's actable, which is then used in the controller methods `create_programming_files` and `destroy_programming_file` in a further nested `Course::Assessment::Answer::Programming::ProgrammingController`.  Loading it there would not make a difference as the `answer_programming` object is already loaded by then.

Reply:
The abstract controller class shouldn't have to be concerned with the subsequent controllers, so placing it here like that doesn't seem to be right. If you wish to authorize it, you should separately authorize this in `Course::Assessment::Answer::Programming::ProgrammingController`.

Reply:
I've thought about that, but if i do so in the `Course::Assessment::Answer::Programming::ProgrammingController`, then i would have to disregard the loading of `answer` and `programming_answer` in `Course::Assessment::Answer::Controller` and `Course::Assessment::Answer::Programming::Controller` all together. If that's the case then doesn't make sense to nest it to begin with since we're not following the class hierarchy.

Reply:
Then the abilities shouldn’t Be declared on `Course::Assessment::Answer`, but the actable model - `Course::Assessment::Answer::Programming`. This will prevent the need for other controllers having to be changed for more nested controllers.